### PR TITLE
Bugfix/max snow layers

### PR DESCRIPTION
**Pull request by arbennett**
This pull request provides a fix for #338. It is bit for bit with the develop branch.

### CODE REVIEW

File: build/source/engine/layerDivide.f90 [hunk 1]
```diff
@@ -259,17 +260,24 @@ subroutine layerDivide(&
 
   ! identify the number of layers to check for need for sub-division
   select case(ix_snowLayers)
-   case(sameRulesAllLayers);    nCheck = nSnow
-   case(rulesDependLayerIndex); nCheck = min(nSnow,4)  ! the depth of the 5th layer, if it exists, does not have a maximum value
+   case(sameRulesAllLayers);    nCheck = min(nSnow, maxSnowLayers-1)
+   case(rulesDependLayerIndex); nCheck = min(nSnow, maxSnowLayers-1)  ! the depth of the 5th layer, if it exists, does not have a maximum value
```
Comment (bartnijssen) [inline build/source/engine/layerDivide.f90]: Fix comment to: `the depth of the last layer (maxSnowLayers), if it exists` ...

File: build/source/engine/layerDivide.f90 [hunk 2]
```diff
@@ -259,17 +260,24 @@ subroutine layerDivide(&
...
 
   ! loop through all layers, and sub-divide a given layer, if necessary
   do iLayer=1,nCheck
+   divideLayer=.false.
 
    ! identify the maximum depth of the layer
    select case(ix_snowLayers)
-    case(sameRulesAllLayers);    zmaxCheck = zmax
+    case(sameRulesAllLayers)
+     if (nCheck >= 99) then
```
Comment (bartnijssen) [inline build/source/engine/layerDivide.f90]: What are the magic numbers `99` and two lines laters `99999.`?

**Comment by bartnijssen** [comment]
@arbennett : Can you just give quick answers to the review questions? I don't quite know what the 99 was for (i understand the 99999. --> veryBig)

File: build/source/engine/layerDivide.f90 (hunk 2 above)
Comment (arbennett) [reply build/source/engine/layerDivide.f90]: 99 was a standin for `maxSnowLayers-1`. The `jrdn1991` snow layering decision sets `maxSnowLayers` to 100. So, the aim is to never split the snowpack into more than `maxSnowLayers`.

**Pull request merged**